File: resource/server/mux.go
package resource_server

import "github.com/aperturerobotics/starpc/srpc"

// NewResourceMux creates a new srpc.Mux and registers services with it.
// Panics if any registration function returns an error.
func NewResourceMux(register ...func(srpc.Mux) error) srpc.Mux {
	mux := srpc.NewMux()
	for _, r := range register {
		if err := r(mux); err != nil {
			panic(err)
		}
	}
	return mux
}

File: sdk/resource/server/mux.ts
import { createMux } from 'starpc'
import type { Mux, Handler } from 'starpc'

// newResourceMux creates a Mux with the given service handlers
// registered.
function newResourceMux(...handlers: Handler[]): Mux {
  const mux = createMux()
  for (const handler of handlers) {
    mux.register(handler)
  }
  return mux
}

export { newResourceMux }
